Doc Tweaks

4Lukaska opened the pull request:
This PR (in tandem with the Mario PR [#2175]) aims to differentiate Mario and Doc a little bit more and remove any weird discrepancies they have (+a bit of code cleanup).

As it is currently, Mario is largely based off of his PM/P+ counterpart which makes his strengths outshine Doc's most of the time (outside of bthrow being better at killing, having easier throw confirms and for some reason having a way faster dash speed). With these changes Mario is now consistently (and more noticeably) more mobile than Doc while having generally weaker kill options which gives Doc a better edge with securing kills.

In general, Doc's moves should now deal 1-3% more damage than Mario's, and most of Doc's combo starters have higher base kb but lower kb growth than Mario's.

There's also a few additional tweaks to both characters to make them feel more complete and more of their "own" character with clearer advantages and disadvantages for each.

Angled moves have also had their hitbox data homogenized.

### **CHANGELOG**

**STATS**
- [-] Dash Speed: 1.7 > 1.52
- [-] Full Hop Height: 30.8805 > 30
- [/] Traction: 0.61 > 0.62
- [+] Shield Size: 9.9u > 10u (now matches Mario's)

Doc now is generally less mobile than Mario. FH height change made to have a consistent ratio with Mario's FH/DJ heights.

**JAB**

Hit 1:
- [+] Damage: 3% > 4%

Hit 2:
- [+] Damage: 2% > 3%

Hit 3:
- [*] Modified motion rate to match Mario’s
- [*] Expression data changed to match Mario’s
- [+] Activity: F3-5 > F3-7 (now matches Mario's)
- [+] Damage: 4% > 6%
- [/] BKB: 60 > 45 (now matches Mario's)
- [/] KBG: 50 > 83 (now matches Mario's)

**DASH ATTACK**

Early:
- [+] Damage: 9.5% > 10%

Late:
- [+] Damage: 8.5% > 9%

**UPTILT**

- [!] Can now reverse hit (now matches Mario's)
- [+] Damage: 8% > 9%

**DTILT**

- [$] Reverted VFX to vanilla to match hitbox change
- [-] Size (Leg/Foot) : 3.5u/4.5u > 3.2u/4.0u (now matches Mario's)
- [-] X Offset (Hip/Foot): -1.0u/6.0u > 0.0u/4.0u (now matches Mario's)
- [/] Z Offset (Foot): 0.5u > 0.0u (now matches Mario's)

Before:
![DMARIO DTILT (OLD)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/05d5183a-2e82-482e-86d0-473c030d05f4)
After:
![DMARIO DTILT (NEW)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/8711ab00-1e81-43d3-91a8-fe34e8870a5c)

**FSMASH**

- [$] SFX Type (Orb): Burn > Electric
- [-] Changed hitbox ids so that the sweetspot has the lowest priority (to match Mario's)
- [!] Can now reverse hit (all hitboxes) (now matches Mario's)
- [/] KBG (Arm/Orb) : 101/93 > 95
- [+] Size (Orb): 4.0u > 5.0u
- [-] X Offset (Orb): 4.5u > 4.0u

High:
- [-] Damage (Arm): 18% > 16%

Mid:
- [/] Damage (Arm/Orb): 17/19% > 16/20%

Low:
- [+] Damage (Orb): 18% > 20%

Before:
![DMARIO FSMASH (OLD)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/9ad43405-301c-4001-b5ff-d62ac39f0661)
After:
![DMARIO FSMASH (NEW)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/19f3e2a7-dfd4-4036-8033-020c7a17668f)

Note: After image does not show arm hitboxes (which didn't have their position or size changed).

**UPSMASH**

- [$] Adjusted VFX
- [$] SFX Type: Kick > Heavy

Early:
- [+] KBG (Early): 95 > 97

Late:
- [$] SFX Level (Air): L > M
- [/] Hitlag Multiplier (Air): 0.8x > 1.0x

Fixed a VFX issue where the upsmash motion arc wouldn't stop when Doc is put into hitlag.

**DSMASH**

- [+] FAF: 48 > 45 (now matches Mario's)

Front:
- [-] Damage (Legs/Thighs): 18%/17% > 17%/16%

Back:
- [-] Damage (Legs): 15% > 14%

**NAIR**

Early:
- [/] Angle: 50 > 361 (now matches Mario's)
- [/] BKB: 25 > 20
- [/] KBG: 95 > 100 (now matches Mario's)

Late:
- [+] BKB: 20 > 25

**FAIR**

- [$] Adjusted VFX (arc now matches Mario's)
- [$] SFX Level (Fist): M > L
- [/] Hitlag Multiplier (Fist): 0.8x > 1.0x (now matches Mario's hitlag increase on sweetspot)
- [+] BKB (Arm/Fist): 30 > 30/35
- [+] KBG : 95 > 100
- [+] Autocancel Window (Early): F1-3 > F1-4 (now matches Mario's)

Early:
- [/] Activity: F16 > F16-17 (now matches Mario's)
- [+] Damage (Arm/Fist): 15%/16% > 16%/17%
- [+] Size (Fist): 3.0u > 4.0u (now matches Mario's)

Before:
![DMARIO FAIR EARLY (OLD)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/8c0970e2-e565-4053-9652-e126bbdb6961)
After:
![DMARIO FAIR EARLY (NEW)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/d7a6eac8-d165-4cf2-beb4-a8c5b08a4d42)

Clean:
- [/] Activity: F17-20 > F18-21 (now matches Mario's)
- [+] X Offset (Fist): 3.2u > 3.4u (now matches Mario's)

Before:
![DMARIO FAIR LATE (OLD)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/8e8c7494-adf3-49d3-9c50-a8b1954deb01)
After:
![DMARIO FAIR LATE (NEW)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/b9076128-1a14-4a83-9430-c98ae76d1291)

Late:
- [/] Activity: F21-22 > F22 (now matches Mario's)
- [+] Damage (Arm/Fist): 14%/15% > 16%/17%

Doc's fair has been slightly beefed up and given consistent strength to match his Melee fair. It should now be a better kill option than Mario's firebrand (the addition of which made Doc's kill power moot).

**BAIR**

- [*] Modified motion rate to match Mario’s
- [$] Adjusted VFX (now matches Mario's)

Early:
- [+] Damage (Legs): 8% > 10%
- [/] Angle (Feet/Legs): 361 > 38/35 (now matches Mario's)
- [/] BKB (Feet/Legs): 20/43 > 25/40 (now matches Mario's)
- [+] KBG (Feet): 105 > 112 (now matches Mario's)

Late:
- [+] Activity: F9-15 > F9-17 (now matches Mario's)
- [+] Damage: 7% > 9%
- [/] Z Offset: 0.0u > 1.0u (now matches Mario's)

Before:
![DMARIO BAIR (OLD)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/b19d5b19-cb90-444a-a018-f16d877940a0)
After:
![DMARIO BAIR (NEW)](https://github.com/HDR-Development/HewDraw-Remix/assets/79385530/b6ed7903-06fd-4bc2-80ea-9e979ea030d0)

**UPAIR**

- [*] Modified motion rate to match Mario’s
- [-] FAF: F30 > F32 (now matches Mario's)

**DAIR**

- [$] Adjusted VFX

**SIDE SPECIAL**

- [*] Removed additional ledge grab check
- [!] Can no longer reverse hit (now matches Mario's)
- [/] Reflection Activity (Projectile Close/Default): F7-22/F9-22 > F6-21
- [+] Damage: 7% > 12%

Reworks reflection activity to match smash4's data rather than Ult's weird over-engineered system.

**UP SPECIAL**

- [+] Damage (Late): 6% > 7%

**DOWN SPECIAL**

- [+] Grounded Heavy Armor Threshold: 5% > 7%
- [/] Swapped hitbox ids of multhits
- [/] Activity (Multihits): F10/14/18/22/26/30 > F10-30 (F4 rehit rate)

.makeit2

4Lukaska committed:
Dr. Electric

ethangarsia517 commented:
The changes to Neutral Special need to be done to Kirby's own ACMD and motion_patch.yaml.

4Lukaska committed:
kerr bee

4Lukaska commented:
> The changes to Neutral Special need to be done to Kirby's own ACMD and motion_patch.yaml.

Done

4Lukaska committed:
up air faf fix

4Lukaska committed:
Pill slipping and chill pill buffs

lol

4Lukaska committed:
chill pill rework

4Lukaska committed:
more chill pill tweaks

4Lukaska committed:
this is the last time im touching chill pill i promise :)

4Lukaska committed:
pill slip refinement

4Lukaska committed:
Pass

4Lukaska committed:
Update aerials.rs

hdr-balance-backroom commented:
Please adjust the following:
- Dash Attack: Remove reverse hit; addressed in dash attack PR
- Fsmash: Revert sourspot removal
- Nair: Revert extra frame of activity
- Uair: Revert everything except the motion rating adjustments
- Megavitamins: Revert all changes
- Chill Pill: Revert all changes, these are now changed in nightly
- Up Special: Revert grounded startup decreases
- Down Special: Revert SDI change, change armor to 7%, swap the hitbox ids of 0 and 2 with 3 and 4
- Revert all grab and throw changes

4Lukaska committed:
Merge branch 'Doc-Changes' of https://github.com/4Lukaska/HewDraw-Remix into Doc-Changes

4Lukaska committed:
Pass 2

4Lukaska committed:
fix

4Lukaska commented:
> Please adjust the following:
> 
>     * Dash Attack: Remove reverse hit; addressed in dash attack PR
> 
>     * Fsmash: Revert sourspot removal
> 
>     * Nair: Revert extra frame of activity
> 
>     * Uair: Revert everything except the motion rating adjustments
> 
>     * Megavitamins: Revert all changes
> 
>     * Chill Pill: Revert all changes, these are now changed in nightly
> 
>     * Up Special: Revert grounded startup decreases
> 
>     * Down Special: Revert SDI change, change armor to 7%, swap the hitbox ids of 0 and 2 with 3 and 4
> 
>     * Revert all grab and throw changes

@hdr-balance-backroom Done. Additionally there has been some minor tweaks to fsmash to accommodate for the sourspot reintroduction and upair has has its faf shifted forwards 2 frames to match Mario's (he can still do 2 up airs in a sh but now it's frame tight). Changelog has been modified to reflect the revisions made.

4Lukaska committed:
oops

hdr-balance-backroom commented:
Please revert the knockback changes to fsmash, the damage increase is enough

4Lukaska commented:
@hdr-balance-backroom Would it be fine if fsmash had the same KB values as Mario's (25 base, 95 growth for both sweet and sourspot)?

4Lukaska committed:
Merge branch 'Doc-Changes' of https://github.com/4Lukaska/HewDraw-Remix into Doc-Changes

4Lukaska committed:
fsmash kb change

4Lukaska commented:
> Please revert the knockback changes to fsmash, the damage increase is enough

@hdr-balance-backroom BKB buff was reverted and KBG for both hits have been changed to 95 (matches Mario's fsmash).
